src/tap/data/ResultSetTableIterator.java

```diff
@@ -731,7 +731,7 @@ else if (colValue instanceof Boolean)
 			else if (colType != null && colValue != null && colType.type == DBDatatype.CHAR && (colType.length == 1 || colType.length <= 0) && colValue instanceof String)
 				colValue = ((String)colValue).charAt(0);
 			// if the column value is a geometrical object, it must be serialized in STC-S:
-			else if (translator != null && colType != null && colType.isGeometry()){
+			else if ((translator != null && colType != null && colType.isGeometry()) || colValue instanceof byte[]) {
```

Comment:
The detection of whether a column is a geometry or not depends of the DB type it has been declared with in the metadata. In your TAPMetadata.columns.datatype, you should have `REGION` or `POINT`, not VARBINARY. Then, it is the mission of `ADQLTranslator.translateGeometryFromDB(...)` to transform this column (a byte[] in your case) into an STC-S expression so that, when selected in an ADQL query, this binary value is seen as a `VARCHAR`.

Applying this additional OR condition will be a real issue for other services having a binary column which is not a geometry. So, I can not accept this modification.

Is there any reason why the datatype of your column is not `REGION` or `POINT`?